### ISSUE TITLE
StackOverflowError when processing platform event

### ISSUE THREAD

**Issue by amortJ** [Adyen/adyen-java-api-library#444]
When the method NotificationHandler.handleMarketpayNotificationJson() is used to translate a platform event in JSON format into a GenericNotification a StackOverflowError occurs.
The issue occurs only for platform events of type ACCOUNT_UPDATED. 

Example to reproduce:
```java
String jsonEvent = "{\"eventDate\":\"2020-10-13T12:57:24+02:00\",\"eventType\":\"ACCOUNT_UPDATED\",\"executingUserKey\":\"foo\",\"live\":true,\"pspReference\":\"1111\n" +
                "2222\",\"content\":{\"pspReference\":\"3333\",\"submittedAsync\":false,\"accountCode\":\"444444\",\"payoutSchedule\":{\"schedule\":\"HOLD\"}}}\n";
NotificationHandler handler = new NotificationHandler();
final GenericNotification notification = handler.handleMarketpayNotificationJson(jsonEvent);
```

```java
Caused by: java.lang.StackOverflowError: null
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[na:na]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1003) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[gson-2.8.6.jar:na]
	at com.adyen.deserializer.MarketPayNotificationMessageDeserializer.deserialize(MarketPayNotificationMessageDeserializer.java:93) ~[adyen-java-api-library-8.0.0.jar:na]
	at com.adyen.deserializer.MarketPayNotificationMessageDeserializer.deserialize(MarketPayNotificationMessageDeserializer.java:45) ~[adyen-java-api-library-8.0.0.jar:na]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1003) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[gson-2.8.6.jar:na]
	at com.adyen.deserializer.MarketPayNotificationMessageDeserializer.deserialize(MarketPayNotificationMessageDeserializer.java:93) ~[adyen-java-api-library-8.0.0.jar:na]
	at com.adyen.deserializer.MarketPayNotificationMessageDeserializer.deserialize(MarketPayNotificationMessageDeserializer.java:45) ~[adyen-java-api-library-8.0.0.jar:na]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1003) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[gson-2.8.6.jar:na]
	at com.adyen.deserializer.MarketPayNotificationMessageDeserializer.deserialize(MarketPayNotificationMessageDeserializer.java:93) ~[adyen-java-api-library-8.0.0.jar:na]
	at com.adyen.deserializer.MarketPayNotificationMessageDeserializer.deserialize(MarketPayNotificationMessageDeserializer.java:45) ~[adyen-java-api-library-8.0.0.jar:na]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1003) ~[gson-2.8.6.jar:na]
...
```

Tested with Adyen Java library 9.0.0

**Comment by andrei-dediu** [comment]
Hey @amortJ,

Thanks for reporting this issue. I have created a ticket internally and we will let you know once we have an update.

Regards,
Andrei.

**Comment by saquibsayyad** [comment]
Hi @amortJ ,

This issue has been fixed in version `10.1.0`. Let us know if you still face any problem.

Kind Regards,
Saquib
Adyen
